plugins/inventory/spacewalk_api.py

```diff
@@ -0,0 +1,183 @@
+
+
+def craft_inventory(group_dict):
+    """Generate hosts and groups as defined in Spacewalk"""
+    # Empty list to hold all members
+    members = []
+    # For each group found in the API call to Spacewalk,
+    for group in group_dict:
+        # List each system
+        member_list = CLIENT.systemgroup.list_systems(KEY, group['name'])
+        # For each system,
+        for member in member_list:
+            # Append the profile name (as this is the editable field in Spacewalk)
+            members.append(member['profile_name'])
+        # Add a new key with group name, and all it's members as the value
+        DYNAMIC_INVENTORY[group['name']] = {'hosts': members, 'vars': {}}
+        # Empty the list for the next group
+        members = []
+    # Also add the required _meta key, for holding hostvars
+    DYNAMIC_INVENTORY['_meta'] = {'hostvars': {}}
+
+
+def host_info():
+    """Gather extra info provided by Spacewalk and use it for hostvars"""
+    # Craft the inventory using the API call from Spacewalk
+    craft_inventory(GROUP_LIST)
+    # For each host,
+    for host in HOST_LIST:
+        # Get some variables
+        id = host['id']
+        name = host['name']
+        # Get useful info from Spacewalk's API to be used as hostvars
+        host_net_devices = CLIENT.system.get_network_devices(KEY, id)
+        host_dmi = CLIENT.system.get_dmi(KEY, id)
+        host_cpu = CLIENT.system.get_cpu(KEY, id)
+        # Create a blank dictionary for each host, which will contain its vars
+        DYNAMIC_INVENTORY['_meta']['hostvars'][name] = {}
+        # Insert the info gathered from the API as hostvars
+        DYNAMIC_INVENTORY['_meta']['hostvars'][name].update({'network': host_net_devices})
+        DYNAMIC_INVENTORY['_meta']['hostvars'][name].update({'dmi': host_dmi})
+        DYNAMIC_INVENTORY['_meta']['hostvars'][name].update({'cpu': host_cpu})
+
+
+# Produce json for ansible-inventory
+def ansible_list(ctx, param, value):
+    """Print entire inventory in JSON"""
+    if not value or ctx.resilient_parsing:
+        return
+    host_info()
+    inventory = json.dumps(DYNAMIC_INVENTORY)
+    click.echo(inventory)
+    ctx.exit()
+
+
+# Get specific host info
+def ansible_host(ctx, param, value):
+    """Print inventory for a specific host"""
+    if not value or ctx.resilient_parsing:
+        return
+    host_info()
+    inventory = json.dumps(DYNAMIC_INVENTORY['_meta']['hostvars'][value])
+    click.echo(inventory)
+    ctx.exit()
+
+
+# Decorated groups for all subcommands
+@click.group()
+@click.version_option('1.0.0')
+# Ansible expected options
+@click.option('--list', is_flag=True, callback=ansible_list,
+              expose_value=False, is_eager=True)
+@click.option('--host', metavar='HOSTNAME', callback=ansible_host,
+              expose_value=True)
+def spacewalk_inventory():
+    """A dynamic inventory script for Ansible that uses the Spacewalk API"""
+    pass
+
+
+if __name__ == '__main__':
```

Comment:
This looks wrong. An inventory plugin should be different, see any of the examples in this directory.

It looks like you wrote an inventory script and placed it in the directory of inventory **plugins**. The scripts are in https://github.com/ansible-collections/community.general/tree/main/scripts/inventory - but please note that we do no longer accept new ones. We only accept inventory plugins (in this directory).

Reply:
Thanks for the good feedback.  It's been a while since I initially wrote this.  It seems a lot has changed with the move to this repo.  I spent some time looking over the docs and getting nowhere on that error so I appreciate the help!

Reply:
The main Ansible repo wasn't accept inventory scripts anymore for quite some time. So that didn't change much :) Inventory plugins didn't change much, the main difference is that you now have to use FQCNs for new plugins, and that you have to adjust some imports.